src/scales/scale.time.js

```diff
@@ -8,34 +8,42 @@ module.exports = function(Chart) {
 
 	var helpers = Chart.helpers;
 	var time = {
-		units: [{
-			name: 'millisecond',
+		millisecond: {
+			size: 1,
 			steps: [1, 2, 5, 10, 20, 50, 100, 250, 500]
-		}, {
-			name: 'second',
+		},
+		second: {
+			size: 1000,
 			steps: [1, 2, 5, 10, 30]
-		}, {
-			name: 'minute',
+		},
+		minute: {
+			size: 60000,
 			steps: [1, 2, 5, 10, 30]
-		}, {
-			name: 'hour',
+		},
+		hour: {
+			size: 3600000,
 			steps: [1, 2, 3, 6, 12]
-		}, {
-			name: 'day',
+		},
+		day: {
+			size: 86400000,
 			steps: [1, 2, 5]
-		}, {
-			name: 'week',
+		},
+		week: {
+			size: 604800000,
 			maxStep: 4
-		}, {
-			name: 'month',
+		},
+		month: {
+			size: 2.628e9,
```

Comment:
These sizes are used to estimate the unit and step size to use, they aren't used for calculating the tick locations. The step size is a multiple of the unit, so a 'unit' of 'month' and 'stepSize' of '3' give ticks which are 3 months apart. The labels are still rounded (by `moment`) to the start of the `unit` so they display correctly.

This very issue is addressed in https://github.com/chartjs/Chart.js/pull/3914/commits/79230683d19f860c6bd2ffc61429b583122c53fb